components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -477,6 +478,19 @@ public static API getAPIForPublishing(GovernanceArtifact artifact, Registry regi
             // set rating
             String artifactPath = GovernanceUtils.getArtifactPath(registry, artifact.getId());
             Resource apiResource = registry.get(artifactPath);
+            Properties properties = apiResource.getProperties();
```

Comment:
There are additional getAPI() methods in APIUtil class that can be used to get the API details. I think you should add this property retrieving section to that apis as well